crates/uv-auth/src/lib.rs

```diff
@@ -1,170 +1,132 @@
+            NetLoc::from(&Url::parse("https://example.com")?)
+        );
 
         // Okay, default port differs (http)
-        assert!(should_retain_auth(
-            &Url::parse("http://example.com:80")?,
-            &Url::parse("http://example.com")?,
-        ));
-        assert!(should_retain_auth(
-            &Url::parse("http://example.com")?,
-            &Url::parse("http://example.com:80")?,
-        ));
+        assert_eq!(
+            NetLoc::from(&Url::parse("http://example.com:80")?),
+            NetLoc::from(&Url::parse("http://example.com")?)
+        );
 
         // Mismatched scheme
-        assert!(!should_retain_auth(
-            &Url::parse("https://example.com")?,
-            &Url::parse("http://example.com")?,
-        ));
+        assert_ne!(
+            NetLoc::from(&Url::parse("https://example.com")?),
+            NetLoc::from(&Url::parse("http://example.com")?)
+        );
 
         // Mismatched scheme, we explicitly do not allow upgrade to https
-        assert!(!should_retain_auth(
-            &Url::parse("http://example.com")?,
-            &Url::parse("https://example.com")?,
-        ));
+        assert_ne!(
+            NetLoc::from(&Url::parse("http://example.com")?),
+            NetLoc::from(&Url::parse("https://example.com")?)
+        );
 
         // Mismatched host
-        assert!(!should_retain_auth(
-            &Url::parse("https://foo.com")?,
-            &Url::parse("https://bar.com")?,
-        ));
+        assert_ne!(
+            NetLoc::from(&Url::parse("https://foo.com")?),
+            NetLoc::from(&Url::parse("https://bar.com")?)
+        );
 
         // Mismatched port
-        assert!(!should_retain_auth(
-            &Url::parse("https://example.com:1234")?,
-            &Url::parse("https://example.com:5678")?,
-        ));
+        assert_ne!(
+            NetLoc::from(&Url::parse("https://example.com:1234")?),
+            NetLoc::from(&Url::parse("https://example.com:5678")?)
+        );
 
         // Mismatched port, with one as default for scheme
-        assert!(!should_retain_auth(
-            &Url::parse("https://example.com:443")?,
-            &Url::parse("https://example.com:5678")?,
-        ));
-        assert!(!should_retain_auth(
-            &Url::parse("https://example.com:1234")?,
-            &Url::parse("https://example.com:443")?,
-        ));
+        assert_ne!(
+            NetLoc::from(&Url::parse("https://example.com:443")?),
+            NetLoc::from(&Url::parse("https://example.com:5678")?)
+        );
+        assert_ne!(
+            NetLoc::from(&Url::parse("https://example.com:1234")?),
+            NetLoc::from(&Url::parse("https://example.com:443")?)
+        );
 
         // Mismatched port, with default for a different scheme
-        assert!(!should_retain_auth(
-            &Url::parse("https://example.com")?,
-            &Url::parse("https://example.com:80")?,
-        ));
```

Comment:
I guess they make less sense with the `NetLoc` implementation?

Reply:
Yeah, because it's now an equality comparison, order doesn't matter.  One of the `NetLoc` is the trusted url, one is the new url, but which side of the equality operation it's on is logically irrelevant.